internal/apps/builder/cnb.go

```diff
@@ -217,6 +226,16 @@ func (b *CNBComponentBuilder) cnbEnv() ([]string, error) {
 		envs = append(envs, "VERSION_PINNING_LIST="+string(versioningJSON))
 	}
 
+	if b.localCacheDir != "" {
+		if exists, err := b.imageExists(ctx, b.ImageOutputName()); err != nil {
+			return nil, err
+		} else if exists {
+			envs = append(envs, "PREVIOUS_APP_IMAGE_URL="+b.ImageOutputName())
+			envs = append(envs, "APP_CACHE_DIR="+cnbCacheDir)
```

Comment:
wouldn't we need to set `APP_CACHE_DIR` either way so we can populate the cache on the first build?